src/api/app/models/workflow/step/link_package_step.rb

```diff
@@ -7,8 +7,9 @@ def call(options = {})
 
     workflow_filters = options.fetch(:workflow_filters, {})
 
-    if scm_webhook.updated_pull_request? && target_package.present?
-      update_subscriptions(target_package, workflow_filters)
+    if scm_webhook.updated_pull_request?
+      create_target_package if target_package.blank?
+      create_or_update_subscriptions(target_package, workflow_filters)
```

Comment:
According to this comment in [Workflow::Step model](https://github.com/openSUSE/open-build-service/blob/2acf0549256926c6b96190bab7a9f054e44ad447/src/api/app/models/workflow/step.rb#L87) there is a plan to remove the method `create_or_update_subscriptions` to use both `create_subscriptions` and `update_subscriptions` instead.

Reply:
hm, could you explain why and who is it planned? Yesterday we were not sure why this `FIXME` was there.

Reply:
When I worked on that code, I was suggested to not mix creating and updating to follow the [KISS principle](https://www.interaction-design.org/literature/topics/keep-it-simple-stupid).
As I was working on LinkPackageStep I postponed the refactoring of BrachPackageStep. I've decided to do it right now :).

Reply:
In the new PR #11601, I refactor the BranchPackageStep the same way I did with LinkPackageStep in #11413

Feel free to base this PR on my changes or not, what is easier for you. I can always fix the conflicts in mine later.

Reply:
We will tackle the refactoring of the BranchPackageStep after this PR is merged.